open-craft/custom-registration-form
# [SE-4603] Admin command to export users info/preferences

# Description

Ticket: [SE-4603](https://tasks.opencraft.com/browse/SE-4603)

This PR adds a management command that can be used to export fields from the User/UserProfile/Registration Extension models into a csv file and upload it to s3.

# Testing instructions

__Prerequisites: get an AWS access key ID, AWS secret key, and an AWS bucket. Run the following commands:__

1. ssh to app server at https://manage.opencraft.com/instance/30000/edx-appserver/20543/
1. `export AWS_ACCESS_KEY_ID="<ID>"`
1. `export AWS_SECRET_ACCESS_KEY="<KEY>"`
1. `cd /edx/app/edxapp`
1. `source edxapp_env`
1. `cd edx-platform`
1. `/edx/bin/manage.edxapp lms email_opt_in_list_to_s3 --s3-bucket-name <BUCKET NAME> --s3-object-name testobj-se-4603 --fields id username email`
1. check the bucket
